network/p2p/p2p.go

```diff
@@ -47,25 +49,27 @@ type listener struct {
 
 // p2pNetwork implements network.Network interface using P2P
 type p2pNetwork struct {
-	ctx           context.Context
-	cfg           *Config
-	listenersLock sync.Locker
-	dv5Listener   iListener
-	listeners     []listener
-	logger        *zap.Logger
-	privKey       *ecdsa.PrivateKey
-	peers         *peers.Status
-	host          p2pHost.Host
-	pubsub        *pubsub.PubSub
+	ctx              context.Context
+	validatorStorage collections.IValidator
+	cfg              *Config
+	listenersLock    sync.Locker
+	dv5Listener      iListener
+	listeners        []listener
+	logger           *zap.Logger
+	privKey          *ecdsa.PrivateKey
+	peers            *peers.Status
+	host             p2pHost.Host
+	pubsub           *pubsub.PubSub
 }
 
 // New is the constructor of p2pNetworker
-func New(ctx context.Context, logger *zap.Logger, cfg *Config) (network.Network, error) {
+func New(ctx context.Context, logger *zap.Logger, validatorStorage collections.IValidator, cfg *Config) (network.Network, error) {
```

Comment:
no need to have the storage in the struct if you only use it to generate topics. 
BTW, the way it's setup it's not dynamic while reading from the contract. You'd have to restart the node to apply new validators?